### PR TITLE
[SPARK-22190][CORE] Add Spark executor task metrics to Dropwizard metrics

### PR DESCRIPTION
**Pull request by LucaCanali**
## What changes were proposed in this pull request?

This proposed patch is about making Spark executor task metrics available as Dropwizard metrics. This is intended to be of aid in monitoring Spark jobs and when drilling down on performance troubleshooting issues.

## How was this patch tested?

Manually tested on a Spark cluster (see JIRA for an example screenshot).

### CODE REVIEW

**Comment by hvanhovell** [comment]
ok to test

**Comment by LucaCanali** [comment]
Thanks @hvanhovell  for pushing the test. I see that the test build threw an error on Scala style tests: the logs report "/Executor.scala:443:0: Whitespace at end of line". However, I cannot reproduce the error when I run scalastyle locally, moreover it appears to me that line Executor.scala:443 is empty (so no extra space there).

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 1]
```diff
@@ -406,6 +406,41 @@ private[spark] class Executor(
...
+          .inc(task.metrics.shuffleWriteMetrics.bytesWritten)
+        executorSource.METRIC_SHUFFLE_RECORDS_WRITTEN
+          .inc(task.metrics.shuffleWriteMetrics.recordsWritten)
+        executorSource.METRIC_INPUT_BYTES_READ
+          .inc(task.metrics.inputMetrics.bytesRead)
+        executorSource.METRIC_INPUT_RECORDS_READ
+          .inc(task.metrics.inputMetrics.recordsRead)
+        executorSource.METRIC_OUTPUT_BYTES_WRITTEN
+          .inc(task.metrics.outputMetrics.bytesWritten)
+        executorSource.METRIC_INPUT_RECORDS_READ
```
Comment (umehrot2) [inline core/src/main/scala/org/apache/spark/executor/Executor.scala]: This has been defined twice in this file. I think what you meant instead is METRIC_OUTPUT_RECORDS_WRITTEN which is missing here.

**Review by umehrot2 — COMMENTED** [review]
-> Any reason for leaving out remote/local blocks fetched metrics from ShuffleReadMetrics ?
-> For the JVM GC Time, it should already be available through JvmSource right ? Is it worth adding here ?

**Comment by LucaCanali** [comment]
Thanks @umehrot2 for the review and comments. Indeed well spotted that I had forgotten a couple of metrics and added one of them twice. This is hopefully fixed with the latest commit. As for your comment on the Spark Executor task metric JVMGCTime, my first reaction is that (1) I think it is good to have this type of info in the context of the executor metrics (2) It does not appear to me that JVMGCTime measured at the executor task level provides a direct duplication of info available in the JVM source (I am available to investigte this in more details in follow-up comments if you wish).

**Review by jiangxb1987 — APPROVED** [review]
LGTM, cc @cloud-fan

**Comment by cloud-fan** [comment]
LGTM, merging to master!

**Pull request closed without merging**